src/_pytest/config/__init__.py

```diff
@@ -112,13 +112,18 @@ def directory_arg(path, optname):
     return path
 
 
-default_plugins = (
+# Plugins that cannot be disabled via "-p no:X" currently.
+essential_plugins = (
     "mark",
     "main",
-    "terminal",
+    "terminal",  # Required to display/handle failures.
```

Comment:
I wonder if this will this affect xdist? https://github.com/pytest-dev/pytest-xdist/blob/master/xdist/remote.py#L252

Reply:
Yeah, certainly.
I had it excluded in the test, because there it failed to report.
It should be not in there then I think.

Reply:
Yeah it seems this will break xdist:

https://github.com/pytest-dev/pytest-xdist/blob/ba434848f5afdfab117191ec304f70c3846185b9/xdist/remote.py#L252

```python
def remote_initconfig(option_dict, args):
    from _pytest.config import Config

    option_dict["plugins"].append("no:terminal")
```

https://github.com/pytest-dev/pytest/blob/0fe114253afbfc2f81491bdc1c3dadbebcb214a2/src/_pytest/config/__init__.py#L192-L197


https://github.com/pytest-dev/pytest/blob/0fe114253afbfc2f81491bdc1c3dadbebcb214a2/src/_pytest/config/__init__.py#L498-L502

Reply:
> It should be not in there then I think.

Agreed. 👍

Reply:
But it must come before "runner" apparently for a recently added test:
```
==================== FAILURES ====================
_ TestPytestPluginManager.test_pluginmanager_ENV_startup _

self = <test_pluginmanager.TestPytestPluginManager object at 0x7faab37cb320>
testdir = <Testdir local('/tmp/pytest-of-user/pytest-839/test_pluginmanager_ENV_startup0')>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7faab1cdd940>

    def test_pluginmanager_ENV_startup(self, testdir, monkeypatch):
        testdir.makepyfile(pytest_x500="#")
        p = testdir.makepyfile(
            """
            import pytest
            def test_hello(pytestconfig):
                plugin = pytestconfig.pluginmanager.get_plugin('pytest_x500')
                assert plugin is not None
        """
        )
        monkeypatch.setenv("PYTEST_PLUGINS", "pytest_x500", prepend=",")
        result = testdir.runpytest(p, syspathinsert=True)
        assert result.ret == 0
>       result.stdout.fnmatch_lines(["*1 passed*"])
E       Failed: nomatch: '*1 passed*'
E           and: '============== test session starts ==============='
E           and: 'platform linux -- Python 3.7.2, pytest-4.3.2.dev151+g6e6d3b74.d20190327, py-1.8.0, pluggy-0.9.0'
E           and: "hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('…/Vcs/pytest/.hypothesis/examples')"
E           and: 'rootdir: /tmp/pytest-of-user/pytest-839/test_pluginmanager_ENV_startup0'
E           and: 'plugins: hypothesis-4.14.0'
E           and: 'collected 1 item'
E           and: ''
E           and: 'test_pluginmanager_ENV_startup.py ...      [100%]'
E           and: ''
E           and: '============ 3 passed in 0.01 seconds ============'
E           and: ''
E       remains unmatched: '*1 passed*'

…/Vcs/pytest/testing/test_pluginmanager.py:268: Failed
-------------- Captured stdout call --------------
============== test session starts ===============
platform linux -- Python 3.7.2, pytest-4.3.2.dev151+g6e6d3b74.d20190327, py-1.8.0, pluggy-0.9.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('…/Vcs/pytest/.hypothesis/examples')
rootdir: /tmp/pytest-of-user/pytest-839/test_pluginmanager_ENV_startup0
plugins: hypothesis-4.14.0
collected 1 item

test_pluginmanager_ENV_startup.py ...      [100%]

============ 3 passed in 0.01 seconds ============
============ short test summary info =============
FAILED testing/test_pluginmanager.py::TestPytestPluginManager::test_pluginmanager_ENV_startup
```

Reply:
Might be worth checking if https://github.com/pytest-dev/pytest/pull/5006 helps with that.